testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -155,6 +155,27 @@ impl BlockMinerThread {
         let mut stackerdbs = StackerDBs::connect(&self.config.get_stacker_db_file_path(), true)
             .expect("FATAL: failed to connect to stacker DB");
 
+        let sort_db = SortitionDB::open(
+            &self.config.get_burn_db_file_path(),
+            true,
+            self.burnchain.pox_constants.clone(),
+        )
+        .expect("FATAL: could not open sortition DB");
+
+        let tip = SortitionDB::get_block_snapshot_consensus(
+            sort_db.conn(),
+            &self.burn_block.consensus_hash,
+        )
+        .expect("FATAL: could not retrieve chain tip")
+        .expect("FATAL: could not retrieve chain tip");
+
+        let reward_set = sort_db
+            .get_preprocessed_reward_set_of(&tip.sortition_id)
+            .expect("FATAL: Error fetching reward set")
+            .expect("FATAL: No reward set found for miner")
+            .known_selected_anchor_block_owned()
+            .expect("FATAL: No reward set found for miner");
```

Comment:
This `.expect()` is reachable in the node's normal mode of operation, since the node may first store a `SelectedAndUnknown` reward set which will get updated to `SelectedAndKnown` once the anchor block is processed.

Reply:
Removed and moved into `gather_signatures`, which will gracefully re-loop if an error occurs